lib/hex_web/assets.ex

```diff
@@ -152,5 +152,5 @@ defmodule HexWeb.Assets do
 
   def tarball_cdn_key(release), do: "tarballs/#{release.package.name}-#{release.version}"
 
-  def tarball_store_key(release), do: "tarballs/#{release.package.name}-#{release.version}.tar"
+  def tarball_store_key(release), do: "tarballs/#{release.package.name}-#{release.version}.tar.gz"
```

Comment:
@ericmj here's the exact fix, the rest of the PR is just refactoring. I guess it was broken all along (since introduction of Assets) :/ I wonder how it even worked, and if we need to do anything else.